### PR TITLE
ad463x: add initial IIO support driver and application

### PR DESCRIPTION
**Pull request by amiclaus**
Add initial IIO support for the ad463x driver.

Provide initial IIO support for the ad463x project.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: drivers/adc/ad463x/iio_ad463x.c [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			struct ad463x_dev *dev);
+{
+	struct iio_ad463x *iio_ad463x;
+
+	iio_ad463x = (struct iio_ad463x *)calloc(1, sizeof(struct iio_ad463x));
+	if (!iio_ad463x)
+		return FAILURE;
+
+	iio_ad463x->ad463x_desc = dev;
+	iio_ad463x->iio_dev_desc = ad463x_iio_desc;
```
Comment (mchindri) [inline drivers/adc/ad463x/iio_ad463x.c:139]: This doesn't need to be saved. You can view it's values anytime

File: drivers/adc/ad463x/iio_ad463x.c [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+static struct iio_channel iio_adc_channels[] = {
+	IIO_AD463x_CHANNEL(0),
+	IIO_AD463x_CHANNEL(1),
+};
+
+struct iio_device ad463x_iio_desc = {
+	.channels = iio_adc_channels,
+	.num_ch = 2,
+	.prepare_transfer = (int32_t (*)())_iio_ad463x_prepare_transfer,
```
Comment (mchindri) [inline drivers/adc/ad463x/iio_ad463x.c:81]: This won't compile, ad463x_iio_desc  needs to be under _iio_ad463x_read_dev

**Review by mchindri — APPROVED** [review]
2 small changes and fix build. Otherwise looks good

**Review by amiclaus — COMMENTED** [review]
changes in v2:
- remove num_ch from init struct.
- make iio_dev_desc extern

**Pull request merged**